cloudinit/sources/DataSourceAzure.py

```diff
@@ -669,7 +669,12 @@ def crawl_metadata(self):
 
     def _is_platform_viable(self):
         """Check platform environment to report if this datasource may run."""
-        return _is_platform_viable(self.seed_dir)
+        chassis_tag = ChassisAssetTag.query_system()
+        if chassis_tag is not None:
+            return True
+
+        # If no valid chassis tag, check for seeded ovf-env.xml
+        return Path(self.seed_dir, "ovf-env.xml").exists()
```

Comment:
Can we move this logic back out into a separate function outside the class named `def is_platform_viable`?
```suggestion
return is_platform_viable(self.seed_dir)
```

The reasons being :
 1: Eventually I'd like to see a standardized functional functional `is_platform_viable` entry point in each DataSource module  which allows for a targeted environment/artifact check of a running environment without looking at cached content or having to initialize a DataSource instance in order to invoke DataSource._is_platform_viable.
 2. Any datasources which define the `is_platform_viable` function will then make unit testing `ds-identify` behavior simpler to ensure the python datasource discovery code and the shell datasource discovery code align for the same use cases.

This also will help us in the future grow additional cloud-init CLI subcommands like `cloud-init check cloud-id` to check the status of the current running environment to see if current IMDS/DMI/environmental artifacts represent if the viable datasource discovery has changed (on a system that becomes sandboxed).

Reply:
Sounds good, will do!

Reply:
Done! I added an optional seed_path so it should work in both cases.